{"payload":{"allShortcutsEnabled":true,"fileTree":{"":{"items":[{"name":"illustration","path":"illustration","contentType":"directory"},{"name":"LICENSE","path":"LICENSE","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"}],"totalCount":3}},"fileTreeProcessingTime":1.493269,"foldersToFetch":[],"reducedMotionEnabled":"system","repo":{"id":633299222,"defaultBranch":"main","name":"Github-test","ownerLogin":"LockenLiu","currentUserCanPush":true,"isFork":false,"isEmpty":false,"createdAt":"2023-04-27T15:49:36.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/103775557?v=4","public":false,"private":true,"isOrgOwned":false},"refInfo":{"name":"main","listCacheKey":"v0:1682592269.0","canEdit":true,"refType":"branch","currentOid":"731034de93942941268100972d9a86f1412c5ce7"},"path":"README.md","currentUser":{"id":103775557,"login":"LockenLiu","userEmail":"lockenL@outlook.com"},"blob":{"rawBlob":null,"colorizedLines":null,"stylingDirectives":null,"csv":null,"csvError":null,"dependabotInfo":{"showConfigurationBanner":null,"configFilePath":null,"networkDependabotPath":"/LockenLiu/Github-test/network/updates","dismissConfigurationNoticePath":"/settings/dismiss-notice/dependabot_configuration_notice","configurationNoticeDismissed":false,"repoAlertsPath":"/LockenLiu/Github-test/security/dependabot","repoSecurityAndAnalysisPath":"/LockenLiu/Github-test/settings/security_analysis","repoOwnerIsOrg":false,"currentUserCanAdminRepo":true},"displayName":"README.md","displayUrl":"https://github.com/LockenLiu/Github-test/blob/main/README.md?raw=true","headerInfo":{"blobSize":"3.71 KB","deleteInfo":{"deletePath":"https://github.com/LockenLiu/Github-test/delete/main/README.md","deleteTooltip":"Delete this file"},"editInfo":{"editTooltip":"Edit this file"},"ghDesktopPath":"x-github-client://openRepo/https://github.com/LockenLiu/Github-test?branch=main&filepath=README.md","gitLfsPath":null,"onBranch":true,"shortPath":"fb863cb","siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2FLockenLiu%2FGithub-test%2Fblob%2Fmain%2FREADME.md","isCSV":false,"isRichtext":true,"toc":[{"level":1,"text":"csMR","anchor":"csmr","htmlText":"csMR"},{"level":2,"text":"Workflow","anchor":"workflow","htmlText":"Workflow"},{"level":2,"text":"Installation","anchor":"installation","htmlText":"Installation"},{"level":2,"text":"Prerequisites","anchor":"prerequisites","htmlText":"Prerequisites"},{"level":3,"text":"Anaconda or Miniconda distribution","anchor":"anaconda-or-miniconda-distribution","htmlText":"Anaconda or Miniconda distribution"},{"level":3,"text":"R dependencies","anchor":"r-dependencies","htmlText":"R dependencies"},{"level":2,"text":"Getting Started","anchor":"getting-started","htmlText":"Getting Started"},{"level":3,"text":"Step 1: Data preparation","anchor":"step-1-data-preparation","htmlText":"Step 1: Data preparation"},{"level":3,"text":"Step 2: Modify the config.yml file","anchor":"step-2-modify-the-configyml-file","htmlText":"Step 2: Modify the config.yml file"},{"level":3,"text":"Step 3: Run csMR","anchor":"step-3-run-csmr","htmlText":"Step 3: Run csMR"},{"level":3,"text":"Step 4: Inspect the output","anchor":"step-4-inspect-the-output","htmlText":"Step 4: Inspect the output"},{"level":2,"text":"Contributors","anchor":"contributors","htmlText":"Contributors"},{"level":2,"text":"Contact","anchor":"contact","htmlText":"Contact"},{"level":2,"text":"Reference","anchor":"reference","htmlText":"Reference"}],"lineInfo":{"truncatedLoc":"99","truncatedSloc":"58"},"mode":"file"},"image":false,"isCodeownersFile":null,"isValidLegacyIssueTemplate":false,"issueTemplateHelpUrl":"https://docs.github.com/articles/about-issue-and-pull-request-templates","issueTemplate":null,"discussionTemplate":null,"language":"Markdown","large":false,"loggedIn":true,"newDiscussionPath":"/LockenLiu/Github-test/discussions/new","newIssuePath":"/LockenLiu/Github-test/issues/new","planSupportInfo":{"repoIsFork":null,"repoOwnedByCurrentUser":null,"requestFullPath":"/LockenLiu/Github-test/blob/main/README.md","showFreeOrgGatedFeatureMessage":null,"showPlanSupportBanner":null,"upgradeDataAttributes":null,"upgradePath":null},"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_dockerfile","dismissStackNoticePath":"/settings/dismiss-notice/publish_stack_from_file","releasePath":"/LockenLiu/Github-test/releases/new?marketplace=true","showPublishActionBanner":false,"showPublishStackBanner":false},"renderImageOrRaw":false,"richText":"<article class=\"markdown-body entry-content container-lg\" itemprop=\"text\"><h1 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-csmr\" class=\"anchor\" aria-hidden=\"true\" href=\"#csmr\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a>csMR</h1>\n<p dir=\"auto\">We introduce the <strong>Cell-Stratified Mendelian Randomization (csMR)</strong> framework to find cell-stratified causality of complex traits/diseases by integrating <strong>GWAS</strong> and <strong>single-cell eQTL data</strong>.</p>\n<h2 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-workflow\" class=\"anchor\" aria-hidden=\"true\" href=\"#workflow\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a>Workflow</h2>\n<h2 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-installation\" class=\"anchor\" aria-hidden=\"true\" href=\"#installation\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a>Installation</h2>\n<p dir=\"auto\">csMR can be downloaded by cloning this repository via the commands:</p>\n<div class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"git clone https://github.com/rhhao/csMR.git\ncd csMR\"><pre>git clone https://github.com/rhhao/csMR.git\n<span class=\"pl-c1\">cd</span> csMR</pre></div>\n<h2 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-prerequisites\" class=\"anchor\" aria-hidden=\"true\" href=\"#prerequisites\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a>Prerequisites</h2>\n<h3 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-anaconda-or-miniconda-distribution\" class=\"anchor\" aria-hidden=\"true\" href=\"#anaconda-or-miniconda-distribution\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a>Anaconda or Miniconda distribution</h3>\n<p dir=\"auto\">In order to void any issues with software versioning, csMR utilises conda environments to automatically install all necessary dependencies. If conda is not present on your system, please install <a href=\"https://www.anaconda.com\" rel=\"nofollow\">anaconda</a> or <a href=\"https://docs.conda.io/en/latest/miniconda.html#\" rel=\"nofollow\">miniconda</a> following the official instructions.</p>\n<p dir=\"auto\">Once conda is ready, users can create an environment to install dependencies that csMR needs though the following commands:</p>\n<div class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"conda install -c conda-forge mamba\nmamba env create --file envs/envpy3.yml\nsource activate csMR\"><pre>conda install -c conda-forge mamba\nmamba env create --file envs/envpy3.yml\n<span class=\"pl-c1\">source</span> activate csMR</pre></div>\n<h3 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-r-dependencies\" class=\"anchor\" aria-hidden=\"true\" href=\"#r-dependencies\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a>R dependencies</h3>\n<p dir=\"auto\">csMR requires R package <code>phenoscanner</code> to filter instrumental in. To install phenoscanner, run the following commands:</p>\n<div class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"if (!require(&quot;devtools&quot;)) { install.packages(&quot;devtools&quot;) }\ndevtools::install_github(&quot;phenoscanner/phenoscanner&quot;, force = TRUE, lib=&quot;&lt;your installation path to conda&gt;/ envs/csMR/lib/R/library&quot;)\"><pre><span class=\"pl-k\">if</span> (<span class=\"pl-k\">!</span>require(<span class=\"pl-s\"><span class=\"pl-pds\">\"</span>devtools<span class=\"pl-pds\">\"</span></span>)) { install.packages(<span class=\"pl-s\"><span class=\"pl-pds\">\"</span>devtools<span class=\"pl-pds\">\"</span></span>) }\n<span class=\"pl-e\">devtools</span><span class=\"pl-k\">::</span>install_github(<span class=\"pl-s\"><span class=\"pl-pds\">\"</span>phenoscanner/phenoscanner<span class=\"pl-pds\">\"</span></span>, <span class=\"pl-v\">force</span> <span class=\"pl-k\">=</span> <span class=\"pl-c1\">TRUE</span>, <span class=\"pl-v\">lib</span><span class=\"pl-k\">=</span><span class=\"pl-s\"><span class=\"pl-pds\">\"</span>&lt;your installation path to conda&gt;/ envs/csMR/lib/R/library<span class=\"pl-pds\">\"</span></span>)</pre></div>\n<p dir=\"auto\"><strong><em>Notice</em>:</strong> in order to successfully run csMR, we suggest to set your temporary R library paths by <code>export R_LIBS=&lt;your installation path to conda&gt;/ envs/csMR/lib/R/library:$R_LIBS</code> before running csMR.</p>\n<h2 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-getting-started\" class=\"anchor\" aria-hidden=\"true\" href=\"#getting-started\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a><strong>Getting Started</strong></h2>\n<h3 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-step-1-data-preparation\" class=\"anchor\" aria-hidden=\"true\" href=\"#step-1-data-preparation\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a><strong>Step 1: Data preparation</strong></h3>\n<p dir=\"auto\">csMR requires summary-level GWAS and QTL data that consist of at least 8 required columns for GWAS and 9 required columns for QTL. Other columns will be ignored during processing.</p>\n<p dir=\"auto\"><strong><em>Notice</em>: the names of required columns MUST be consistent with the following headers</strong>, while the order can be inconsistent. The required data format is presented as follows:</p>\n<p dir=\"auto\">GWAS summary stats:</p>\n<table>\n<thead>\n<tr>\n<th align=\"center\">SNP</th>\n<th align=\"center\">A1</th>\n<th align=\"center\">A2</th>\n<th align=\"center\">MAF</th>\n<th align=\"center\">BETA</th>\n<th align=\"center\">SE</th>\n<th align=\"center\">P</th>\n<th align=\"center\">N</th>\n</tr>\n</thead>\n<tbody>\n<tr>\n<td align=\"center\">rs1</td>\n<td align=\"center\">A</td>\n<td align=\"center\">T</td>\n<td align=\"center\">0.125</td>\n<td align=\"center\">0.166</td>\n<td align=\"center\">0.178</td>\n<td align=\"center\">0.734</td>\n<td align=\"center\">480000</td>\n</tr>\n<tr>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n</tr>\n<tr>\n<td align=\"center\">rs256</td>\n<td align=\"center\">C</td>\n<td align=\"center\">G</td>\n<td align=\"center\">0.201</td>\n<td align=\"center\">0.037</td>\n<td align=\"center\">0.079</td>\n<td align=\"center\">0.349</td>\n<td align=\"center\">480000</td>\n</tr>\n</tbody>\n</table>\n<p dir=\"auto\">QTL summary stats:</p>\n<table>\n<thead>\n<tr>\n<th align=\"center\">SNP</th>\n<th align=\"center\">GENE</th>\n<th align=\"center\">A1</th>\n<th align=\"center\">A2</th>\n<th align=\"center\">MAF</th>\n<th align=\"center\">BETA</th>\n<th align=\"center\">SE</th>\n<th align=\"center\">P</th>\n<th align=\"center\">N</th>\n</tr>\n</thead>\n<tbody>\n<tr>\n<td align=\"center\">rs1</td>\n<td align=\"center\">geneY</td>\n<td align=\"center\">A</td>\n<td align=\"center\">T</td>\n<td align=\"center\">0.359</td>\n<td align=\"center\">-0.039</td>\n<td align=\"center\">0.094</td>\n<td align=\"center\">0.082</td>\n<td align=\"center\">215</td>\n</tr>\n<tr>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n<td align=\"center\">…</td>\n</tr>\n<tr>\n<td align=\"center\">rs256</td>\n<td align=\"center\">geneX</td>\n<td align=\"center\">C</td>\n<td align=\"center\">G</td>\n<td align=\"center\">0.229</td>\n<td align=\"center\">0.064</td>\n<td align=\"center\">0.012</td>\n<td align=\"center\">0.779</td>\n<td align=\"center\">215</td>\n</tr>\n</tbody>\n</table>\n<h3 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-step-2-modify-the-configyml-file\" class=\"anchor\" aria-hidden=\"true\" href=\"#step-2-modify-the-configyml-file\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a>Step 2: Modify the <code>config.yml</code> file</h3>\n<p dir=\"auto\">Before running csMR, a configuration file needs to be modified to specify the input data and advanced parameters. An example is provided in the <code>config.yml</code> file that will process the example data located in <code>example </code>directory.</p>\n<h3 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-step-3-run-csmr\" class=\"anchor\" aria-hidden=\"true\" href=\"#step-3-run-csmr\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a>Step 3: Run csMR</h3>\n<p dir=\"auto\">Now you have everything csMR needs, then you can run the analysis workflow by:</p>\n<div class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"snakemake -s work_flow.snakefile --configfile config.yml -j\"><pre>snakemake -s work_flow.snakefile --configfile config.yml -j</pre></div>\n<p dir=\"auto\">We recommend running with <code>-j</code> as it will use all available cores. Specifying <code>-j 4</code> will use up to 4 cores.</p>\n<h3 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-step-4-inspect-the-output\" class=\"anchor\" aria-hidden=\"true\" href=\"#step-4-inspect-the-output\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a>Step 4: Inspect the output</h3>\n<p dir=\"auto\">The results of colocalization and MR will be placed in <code>&lt;your output dir&gt;/COLCO</code>and <code>&lt;your output dir&gt;/MR</code> respectively.</p>\n<h2 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-contributors\" class=\"anchor\" aria-hidden=\"true\" href=\"#contributors\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a><strong>Contributors</strong></h2>\n<p dir=\"auto\">Ruo-Han Hao (Xi’an Jiaotong University, China)</p>\n<p dir=\"auto\">Feng Jiang (Xi’an Jiaotong University, China)</p>\n<p dir=\"auto\">Tian-Pei Zhang (Xi’an Jiaotong University, China)</p>\n<h2 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-contact\" class=\"anchor\" aria-hidden=\"true\" href=\"#contact\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a><strong>Contact</strong></h2>\n<p dir=\"auto\">Please create an issue on the github repo if you encounter any problems. You can also contact the developers through email: <a href=\"mailto:ruohanhao@xjtu.edu.cn\">ruohanhao@xjtu.edu.cn</a>.</p>\n<h2 tabindex=\"-1\" dir=\"auto\"><a id=\"user-content-reference\" class=\"anchor\" aria-hidden=\"true\" href=\"#reference\"><svg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"><path d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"></path></svg></a><strong>Reference</strong></h2>\n</article>","renderedFileInfo":null,"tabSize":8,"topBannersInfo":{"overridingGlobalFundingFile":false,"globalPreferredFundingPath":null,"repoOwner":"LockenLiu","repoName":"Github-test","showInvalidCitationWarning":false,"citationHelpUrl":"https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-citation-files","showDependabotConfigurationBanner":null,"actionsOnboardingTip":null},"truncated":false,"viewable":true,"workflowRedirectUrl":null,"symbols":{"timedOut":false,"notAnalyzed":true,"symbols":[]}},"csrf_tokens":{"/LockenLiu/Github-test/branches":{"post":"fDCuqD6rMoBVgg54GtVLSJ37OqfGQwFsxK_HK7qctQEiWhtjKFi_xL4cWNaMaq64kpwk-jFfNE7RSeKSXUom6A"}}},"title":"Github-test/README.md at main · LockenLiu/Github-test","locale":"en"}